mcode/mcode-coverage-checker
# Adding Outcome coverage checking, changing Patient coverage checking to match

# Description
Issue: STEAM-845

This PR adds coverage checking for Outcome resources and changes Patient coverage checking to output in the new format 

## Important Changes
Both patient and outcome coverage checking conform to a new output format, which looks like this:
```
{
  "section": "section name",
  "data": [
    {
      "profile": "profile name",
      "coverage": [
        {
          "resourceID": "a-resource-id",
          "data": {
            "property name": {
              "covered": true
            },
            "another property name": {
              "covered": false
            }
          }
        },
        {
          "resourceID": "another-resource-id",
          "data": {
            "property name": {
              "covered": true
            },
            "another property name": {
              "covered": true
            }
          }
        },
        ...
      ]
    },
    ...
  ]
}
```

Coverage checking has been added for the outcome section of the mCODE diagram in `outcomeCoverage.js`. As we discussed, Body Structure Identifier, despite being a profile on the diagram, is treated as a property of Tumor resources. Another thing we discussed, Tumor Specimen had no properties listed on the diagram so I checked for Type (which I think we decided was a reasonable thing to check for?), definitely let me know if there are different and/or other properties we want to check for.

Patient coverage checking, in `patientCoverage.js`, also has one additional  field added, Death Date, which was in the outcome section on the diagram, but added to the Patient section for simplicity

Removed the `getBodyStructureIdentifier()` function from `resourceUtils.js` for reasons mentioned above about Body Structure Identifier

Tests added and updated for Outcome and Patient in `patientCoverage.test.js` and `outcomeCoverage.test.js`. I've also frankenstein-ed together a few different example resources in `fullBundle.js` and the respective section test bundles, so all the references may not match up, but I think it should work for the purposes of coverage testing.

## Testing Recommendations
- Ensure that new tests cover sufficient cases and pass
- Check that the new output format makes sense
- Make sure that each field has a proper true/false value when it is covered or not

# Checklists

**Submitter:**

@ACCT1 :
